Fill out the systemd-nspawn support

terminalmage opened the pull request:
So far I have added a couple additional functions to manage and get the
container state, as well as a set of functions to run commands within
containers, and interfaces to `machinectl pull-*`.

**Still to add (in a future PR):**
- Function to run salt-bootstrap in a container
- Function to clone a container.
- Functions for `machinectl copy-to` and `machinectl copy-from`

terminalmage committed:
Fix nspawn.bootstrap

locals(), in this context, is local to the bootstrap function, needed to
be globals(). Also was missing parenthesis.

terminalmage committed:
add nspawn.stop

terminalmage committed:
Improve config.get docstring

terminalmage committed:
Fill out some functonality in nspawn module

Add functions to determine and manage container state, as well as
functions to run commands within containers.

s0undt3ch commented on salt/modules/nspawn.py:
```diff
@@ -1,19 +1,44 @@
+these containers.
+
+.. __: http://www.freedesktop.org/software/systemd/man/systemd-nspawn.html
+
```
Lovely!!!! Unnamed links! Go @terminalmage!

terminalmage committed:
Add function to get systemd version

terminalmage committed:
Add RST file for nspawn

terminalmage committed:
Use new utils function to check systemd version

terminalmage committed:
Fill out more functionality in nspawn module

terminalmage committed:
Update test_valid_docs for nspawn compatibility funcs

terminalmage committed:
Update nspawn top-level docstring

This commit reflects the different default location for containers in
systemd >= 219.

thatch45 commented:
Once tests pass we can merge in this iteration

terminalmage commented:
Yeah I should be able to get the salt-bootstrap function (already partially written) in a separate PR.

terminalmage committed:
Fix lint failures

terminalmage commented:
Just pushed some lint fixes.

terminalmage committed:
Add missing CLI examples

terminalmage commented:
Also just pushed fixes for missing CLI examples caught in the Cent7 tests.

terminalmage committed:
Add compatibility function "remove"

For people using a mix of LXC and nspawn containers

terminalmage committed:
Add nspawn.remove function

Also adds nspawn.destroy as an alias, for those who are used to managing
LXC containers.

terminalmage committed:
docker index URL is a required parameter
